### ISSUE TITLE
Alpine 3.11

### ISSUE THREAD

**Issue by J0WI** [nginxinc/docker-nginx#380]
Please add packages for [Alpine 3.11](https://alpinelinux.org/posts/Alpine-3.11.0-released.html).

**Comment by thresheek** [comment]
Hello and thank you @J0WI for the heads up.

We're in a middle of infrastructure move these days and it will take some time to prepare gold images for Alpine 3.11 to build our releases on.  I'd expect them to arrive in the next month.

**Comment by stephankaag** [comment]
Can you share a status update @thresheek? :-)

**Comment by thresheek** [comment]
Thanks for pinging!

Yep - I've just finished adding alpine 3.11 into our build infra and waiting for the next night's build and test results.  Once they're in, I'll build and publish the packages for the official releases.  Hopefully it'll be there early next week.

**Comment by thresheek** [comment]
The packages are ready and published on nginx.org to enjoy.  I'll wait for new releases of nginx to move the official images over to 3.11.  Not going to replace the already released tags with the newer content.

**Comment by thresheek** [comment]
OK, now nginx:1.17.9-alpine uses alpine:3.11.

I'll wait for stable branch to receive a new release and then I'll move it to alpine:3.11 as well.

**Comment by MatMaul** [comment]
@thresheek I can still see 3.10 in https://github.com/nginxinc/docker-nginx/blob/master/mainline/alpine/Dockerfile , did I miss something ?
Thanks.

**Comment by thresheek** [comment]
Oh, it seems I forgot to push the actual change.  My bad. :/

**Comment by esemeniuc** [comment]
@thresheek looks like 3.10 still?

```bash
➜  client git:(master) ✗ docker pull nginx:alpine                
alpine: Pulling from library/nginx
Digest: sha256:abe5ce652eb78d9c793df34453fddde12bb4d93d9fbf2c363d0992726e4d2cad
Status: Image is up to date for nginx:alpine
docker.io/library/nginx:alpine
➜  client git:(master) ✗ docker run -it --rm nginx:alpine cat /etc/apk/repositories
http://dl-cdn.alpinelinux.org/alpine/v3.10/main
http://dl-cdn.alpinelinux.org/alpine/v3.10/community
➜  client git:(master) ✗ 
```

**Comment by MilesWells** [comment]
@thresheek Hello! It seems both nginx:1.17.9-alpine and nginx:alpine are still on 3.10. The bump to the 3.11 Alpine branch includes a bump for the nodejs package to Node 12 LTS. 

For now I'm going to use nginx:alpine and manually install Node 12, but in the future I'd like to replace it with a simple apk add!

**Comment by thresheek** [comment]
Hi @MilesWells, hope all is good.

The desire to use the latest version is understood, and rest assured, nginx:alpine will be based on alpine:3.11 for the next release.

**Comment by jf423** [comment]
Hi @thresheek , 
May I ask when will be the next release for the alpine:3.11 or could we have a beta version for try first?

**Comment by thresheek** [comment]
The beta version would be just change the first line in the Dockerfile from "FROM alpine:3.10" to "FROM alpine:3.11".

Not sure when the next release would happen, I expect mid April.

**Comment by J0WI** [comment]
>  I'll wait for new releases of nginx to move the official images over to 3.11. Not going to replace the already released tags with the newer content.

It's not clear to me why you postpone this to a nginx release. There is not much to care about in the base image and nginx itself doesn't have any complex dependencies. If so, images in the library tend to have aliases with suffixes like `-alpine3.10` and `-alpine3.11` (e.g. [python](https://hub.docker.com/_/python) or [ruby](https://hub.docker.com/_/ruby)).

I've filled #396 for today's release.

**Comment by tn-osimis** [comment]
FWIW there are some Alpine packages commonly used by nginx users that could really benefit from such rebases asynchronous to nginx releases; in our case we're waiting on 3.11 for up-to-date versions of Certbot (given its use alongside nginx with its nginx plugin, I'm assuming we're not the only ones) which we currently set up very conveniently on top of this official nginx image (thanks for work on it!). We totally respect a maintainer's wish _not_ to provide these additional images (it's entirely your prerogative and we could definitely temporarily fork if we need to, which we just might end up doing depending on our options and requirements), but I just thought I'd chime in to indicate that at the very least, if the effort was to be put in, it would be _greatly_ appreciated! ;)

(The Docker Hub official image is, after all, kind of a [focal point](https://en.wikipedia.org/wiki/Focal_point_(game_theory)) for many.)

**Comment by thresheek** [comment]
Thanks for your input @tn-osimis - that's exactly the input that can motivate me in being less conservative in version updates, which I tend to be since there are oh so many users of nginx and any change can introduce changes in behaviour at best and plain and simple errors at worst.

That being said, all current alpine nginx images now are based off alpine 3.11, so I'm closing that issue.

Thank you everyone.

**Comment by tn-osimis** [comment]
@thresheek Oh wow, that feeling; that was silly. (As you can undoubtedly imagine I was set on waiting for this issue and didn't even bother to double-check again recently -- sorry for the noise and thanks for the heads up!)
